app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java

```diff
@@ -57,8 +61,10 @@
 
   HomePresenterImpl(BaseContract.View view) {
     this.view = (HomeContract.View) view;
-    preferenceRepository = PreferenceRepositoryImpl.get();
 
+    geocodingAPIInterface = GeocodingAPIClient.getClient().create(GeocodingAPIInterface.class);
+
+    preferenceRepository = PreferenceRepositoryImpl.get();
     preferenceRepository.onPreferenceChangeListener(
```

Comment:
refactor the change listener as there is no way to know the old temperature

Reply:
:ok: